### PR TITLE
Enable epel for packit build & testing farm

### PR DESCRIPTION
**Pull request by psss**

### CODE REVIEW

**Comment by thrix** [comment]
tests are failing because fmf is outdated seems:
```
_________________ ERROR collecting tests/unit/test_convert.py __________________
tests/unit/test_convert.py:5: in <module>
    import tmt.cli
tmt/cli.py:362: in <module>
    _test_templates = listed(tmt.templates.TEST, join='or')
E   TypeError: listed() got an unexpected keyword argument 'join'
```

join was added in 0.9-1

**Comment by psss** [comment]
Yes, the fresh `fmf` packages are in [bodhi](https://bodhi.fedoraproject.org/updates/?packages=fmf). Please add karma :)

**Comment by TomasTomecek** [comment]
(I just checked the build failure and packit is *not* to blame here: your tests failed during the RPM build :)

**Comment by thrix** [comment]
nice, we are a bit further ... I am trying to enable epel-8 chroot now ... 

As for other errors, I need to check what is the problem. I guess for the F-29 we currently are not able to skip plans which are not runnable ...

**Comment by thrix** [comment]
nice, epel-8 is now tested on centos-stream ... question becomes: why we cannot install it :(

**Comment by thrix** [comment]
ah right, we need to actually install epel8 on centos stream machine

**Comment by thrix** [comment]
Seems to me we can revisit this later, I do not think we need to do anything else right now.

**Comment by thrix** [comment]
oh nm, reopening :D this should really work now

**Comment by thrix** [comment]
rebasing and rebuilding ...

**Comment by thrix** [comment]
> rebasing and rebuilding ...

ah rebase already done :) ok, let's test this on FRESH centos stream updated recently

**Comment by psss** [comment]
This causes jobs to fail on Fedora because there is no `epel-release` package there. Shall we leave this for later? Or add some condition logic?

File: ansible/packages.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 - hosts: all
   tasks:
+  - name: Install EPEL
+    dnf:
+      state: present
+      name: epel-release
```
Comment (thrix) [inline ansible/packages.yml]: @psss we just need a conditional here ...

Comment (thrix) [reply ansible/packages.yml]: will add it and retry

**Comment by thrix** [comment]
@TomasTomecek  are the errors not forwarded our problem? Seems we are sending it ....

**Comment by thrix** [comment]
oh well, for why this is failing, for epel targets we should also enable epel before installing stuff .... I need to fix that

**Comment by thrix** [comment]
Ok, now tmt-all cannot be installed :) you know why @psss

**Comment by thrix** [comment]
I fixed the previous problem ...

**Comment by psss** [comment]
> Ok, now tmt-all cannot be installed :) you know why @psss

Oh, now I know...

    Problem: cannot install the best candidate for the job
    nothing provides vagrant needed by tmt-all-0.8-1.el8.noarch
    nothing provides vagrant-libvirt needed by tmt-all-0.8-1.el8.noarch

@pvalena, any plans to include those in epel?

**Comment by thrix** [comment]
> > Ok, now tmt-all cannot be installed :) you know why @psss
> 
> Oh, now I know...
> 
> ```
> Problem: cannot install the best candidate for the job
> nothing provides vagrant needed by tmt-all-0.8-1.el8.noarch
> nothing provides vagrant-libvirt needed by tmt-all-0.8-1.el8.noarch
> ```
> 
> @pvalena, any plans to include those in epel?

Maybe we could use some copr repository added in playbook ...

**Comment by thrix** [comment]
Btw the issue with errors not displayed seems to be a bug/regression in Packit: https://github.com/packit-service/packit-service/issues/387

**Comment by pvalena** [comment]
> > > Ok, now tmt-all cannot be installed :) you know why @psss
> > 
> > 
> > Oh, now I know...
> > ```
> > Problem: cannot install the best candidate for the job
> > nothing provides vagrant needed by tmt-all-0.8-1.el8.noarch
> > nothing provides vagrant-libvirt needed by tmt-all-0.8-1.el8.noarch
> > ```
> > 
> > 
> > @pvalena, any plans to include those in epel?
> 
> Maybe we could use some copr repository added in playbook ... in the mean time ...

Sorry for the troubles. Can you try please `pvalena/vagrant` COPR repo?

EPEL does not seem realistic at least at this point (you can try to push for it get 'supported'). EPEL would prevent me doing upgrades, which is how we mostly deal with errors in Fedora-rawhide (and I can't maintain separate stream and backport issues... as it's too much work). I was thinking of some epel-playground, or if it could be in some `epel-module` that would be great...

**Comment by thrix** [comment]
@pvalena copr is fine ... I just need to move the artifact installation before prepare phase for now ...

**Comment by thrix** [comment]
@pvalena, seems I also need other copr repos?

with pvalena/vagrant I get this:

Error: 
 Problem: package tmt-all-0.8-1.el8.noarch requires vagrant, but none of the providers can be installed
  - cannot install the best candidate for the job
  - nothing provides rubygem(i18n) < 2.0 needed by vagrant-2.2.7-1.el8.1.noarch
  - nothing provides rubygem(bcrypt_pbkdf) needed by vagrant-2.2.7-1.el8.noarch
  - nothing provides rubygem(ed25519) needed by vagrant-2.2.7-1.el8.noarch
  - nothing provides rubygem(i18n) < 2.0 needed by vagrant-2.2.7-1.el8.noarch

I guess we need more vagrant repos ;)

**Comment by pvalena** [comment]
> I guess we need more vagrant repos ;)

Yes, I know about that and fixing it as of now asap (I thought I've already pushed the fix, but apparently not).

**Comment by TomasTomecek** [comment]
anything we can help here?

**Comment by thrix** [comment]
@TomasTomecek 

> anything we can help here?

I think we are fine here, the issue we were discussing is our problem :)

**Comment by psss** [comment]
Seems we [do not have](https://console-testing-farm.apps.ci.centos.org/pipeline/76f977fb-3037-47bd-8a04-44f0eeb27133) `beakerlib` in the `epel8` repo:

```
# dnf -y install beakerlib git beakerlib-librariesLast metadata expiration check: 0:00:17 ago on Wed 13 May 2020 01:51:29 PM UTC.No match for argument: beakerlibError: Unable to find a match: beakerlib
```
@sopos, could you please build it? This is blocking beakerlib test execution on epel8 in testing farm. Thanks.

**Pull request merged**